cmd/cloud_sql_proxy/proxy.go

```diff
@@ -221,62 +221,77 @@ var validNets = func() map[string]bool {
 
 func parseInstanceConfig(dir, instance string, cl *http.Client) (instanceConfig, error) {
 	var ret instanceConfig
-	eq := strings.Index(instance, "=")
-	if eq != -1 {
-		spl := strings.SplitN(instance[eq+1:], ":", 3)
-		ret.Instance = instance[:eq]
-
-		switch len(spl) {
-		default:
-			return ret, fmt.Errorf("invalid %q: expected 'project:instance=tcp:port'", instance)
-		case 2:
-			// No "host" part of the address. Be safe and assume that they want a
-			// loopback address.
-			ret.Network = spl[0]
-			addr, ok := loopbackForNet[spl[0]]
-			if !ok {
-				return ret, fmt.Errorf("invalid %q: unrecognized network %v", instance, spl[0])
+	args := strings.Split(instance, "=")
+	if len(args) > 2 {
+		return instanceConfig{}, fmt.Errorf("invalid instance argument: must be either form - `<instance_connection_string>` or `<instance_connection_string>=<options>`; invalid arg was %q", instance)
+	}
+	// Parse the instance connection name - everything before the "=".
+	ret.Instance = args[0]
+	proj, _, name := util.SplitName(ret.Instance)
+	if proj == "" || name == "" {
+		return instanceConfig{}, fmt.Errorf("invalid instance connection string: must be in the form `project:region:instance-name`; invalid name was %q", args[0])
+	}
+	// Parse the instance options - everything before the "=".
+	eM := "invalid instance options: must be in the form `unix:/path/to/socket`, `tcp:port`, `tcp:host:port`; invalid option was %q"
```

Comment:
what is eM? error message? please inline where it's used (L254)

Reply:
It's used on L241 and L254 - I would prefer to keep the same error for both lines

Reply:
ok, please give it a better name or a concrete type, like:

```
type fooError {
  invalidOption string
}

func (e fooError) Error() string {
  return fmt.Sprintf(...)
}
```